src/Sql/Vault/dbo/Stored Procedures/SecurityTask/SecurityTask_ReadByUserIdStatus.sql

```diff
@@ -0,0 +1,45 @@
+CREATE PROCEDURE [dbo].[SecurityTask_ReadByUserIdStatus]
+    @UserId UNIQUEIDENTIFIER,
+    @Status AS [dbo].[SecurityTaskStatusArray] READONLY
+AS
+BEGIN
+    SET NOCOUNT ON
+
+    SELECT
+        ST.*
+    FROM
+        [dbo].[SecurityTaskView] ST
+    INNER JOIN
+        [dbo].[OrganizationUserView] OU ON OU.[OrganizationId] = ST.[OrganizationId]
+    INNER JOIN
+        [dbo].[Organization] O ON O.[Id] = ST.[OrganizationId]
+    LEFT JOIN
+        [dbo].[CipherView] C ON C.[Id] = ST.[CipherId]
+    LEFT JOIN
+        [dbo].[CollectionCipher] CC ON CC.[CipherId] = C.[Id] AND C.[Id] IS NOT NULL
+    LEFT JOIN
+        [dbo].[CollectionUser] CU ON CU.[CollectionId] = CC.[CollectionId] AND CU.[OrganizationUserId] = OU.[Id] AND C.[Id] IS NOT NULL
+    LEFT JOIN
+        [dbo].[GroupUser] GU ON GU.[OrganizationUserId] = OU.[Id] AND CU.[CollectionId] IS NULL AND C.[Id] IS NOT NULL
+    LEFT JOIN
+        [dbo].[CollectionGroup] CG ON CG.[GroupId] = GU.[GroupId] AND CG.[CollectionId] = CC.[CollectionId]
+    WHERE
+        OU.[UserId] = @UserId
+        AND OU.[Status] = 2 -- Ensure user is confirmed
+        AND O.[Enabled] = 1
+        AND (
+            ST.[CipherId] IS NULL
+            OR (
+                C.[Id] IS NOT NULL
+                AND (
+                    CU.[Manage] = 1
+                    OR CG.[Manage] = 1
+                )
+            )
+        )
+        AND (
+            NOT EXISTS (SELECT 1 FROM @Status)
```

Comment:
❓ Could we simplify this SQL by having the query or repository method ensure the `Status` parameter always has default values if none are provided? I believe it would allow us to simply always filter by `ST.[Status] IN (SELECT [Value] FROM @Status)` and avoid the `NOT EXISTS (SELECT 1 FROM @Status)`.

Reply:
Won't we lose the ability to retrieve all tasks regardless of their status if we use this approach?

Reply:
My thinking was we would have the repository method always provide a default value of every status type so that all tasks could be included. With that said, I don't feel too strongly about it either way, just thought it may make the SQL a little easier to follow.